client/fingerprint/env_aws.go

```diff
@@ -104,24 +108,25 @@ func (f *EnvAWSFingerprint) Fingerprint(request *FingerprintRequest, response *F
 	}
 
 	for k, unique := range keys {
-		resp, err := ec2meta.GetMetadata(k)
-		v := strings.TrimSpace(resp)
-		if v == "" {
-			f.logger.Debug("read an empty value", "attribute", k)
-			continue
-		} else if awsErr, ok := err.(awserr.RequestFailure); ok {
-			f.logger.Debug("could not read attribute value", "attribute", k, "error", awsErr)
+		resp, err := imdsClient.GetMetadata(ctx, &imds.GetMetadataInput{
```

Comment:
Outside the scope for this PR but I wonder if there would be any benefit in the future from not passing individual keys to the metadata call, and rather get the whole response and process that. The benefit would be a single API call at the cost of more complex response processing.

Reply:
This would be nice, but I'm not sure it's possible with the current IMDS.  If you pass in an empty path, you just get a list of available paths, not all the instance metadata.